File: leetcode/1894-merge-strings-alternately/merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string word3="";
        int l1=word1.length();
        int l2=word2.length();
        int l=0;
        int i=0;
        int j=0;
        while(i < l1 && j < l2){
            if(l%2==0){
            word3+=word1[i++];
            }
            else{
                word3+=word2[j++];
            }
            l++;
        }
        while(i<l1){
            word3+=word1[i++];
        }
        while(j<l2){
            word3+=word2[j++];
        }
        return word3;
    }

};

File: leetcode/1894-merge-strings-alternately/merge-strings-alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        l1 = len(word1)
        l2 = len(word2) 
        i,j = 0,0
        res = ""
        while i < l1 and j < l2:
            res += (word1[i]+word2[j])
            i += 1
            j += 1
        print(word2[j:])
        if i < l1:
            res += (word1[i:])
        
        if j < l2:
            res += (word2[j:])
        return res